components/support/rc_crypto/src/contentsignature.rs

```diff
@@ -0,0 +1,305 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::str;
+
+use crate::error::*;
+use crate::signature;
+
+fn decode_root_hash(input: &str) -> Result<Vec<u8>> {
+    let bytes_hex = input.split(':');
+
+    let mut result: Vec<u8> = vec![];
+    for byte_hex in bytes_hex {
+        let byte = match hex::decode(&byte_hex) {
+            Ok(v) => v,
+            Err(_) => return Err(ErrorKind::RootHashFormatError(input.to_string()).into()),
+        };
+        result.extend(byte);
+    }
```

Comment:
If you wanted to mess around with Rust iterators, this would be a good opportunity to use `collect`. It's possible to [`collect` into a `Result<Vec<T>>`](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect) and get the early-termination-on-error thing for free, something along the lines of:

```
fn decode_root_hash(input: &str) -> Result<Vec<u8>> {
    input.split(':').map(|byte_hex| {
        hex::decode(&byte_hex).map_err(|_| {
            ErrorKind::RootHashFormatError(input.to_string()).into()
        })
    }).collect()
}
```

(The more explicit version is also fine if you *don't* want to mess around with Rust iterators).

Reply:
I could not figure out how to flatten items and collect results, so I ended up writing an explicit loop :man_shrugging:

Reply:
ah well, it was worth a try :-)